src/src/Commands/Tasks.cpp

```diff
@@ -39,12 +41,42 @@ bool validTaskVars(struct EventStruct *event, taskIndex_t& taskIndex, unsigned i
   return true;
 }
 
+/**
+ * parse TaskName/TaskValue when not numeric for task name and value name and validate values
+ */
+bool validateAndParseTaskValueArguments(struct EventStruct * event, const char *Line, taskIndex_t &taskIndex, unsigned int &varNr)
+{
+  if (!validTaskVars(event, taskIndex, varNr) || (event->Par2 <= 0 || event->Par2 >= VARS_PER_TASK))  // Extra check required because of shortcutting in validTaskVars()
+  { 
+    String taskName;
+    taskIndex_t tmpTaskIndex = taskIndex;
+    if ((event->Par1 <= 0 || event->Par1 >= INVALID_TASK_INDEX) && GetArgv(Line, taskName, 2)) {
+      tmpTaskIndex = findTaskIndexByName(taskName);
+      if (tmpTaskIndex != INVALID_TASK_INDEX) {
+        event->Par1 = tmpTaskIndex + 1;
+      }
+    }
+    String valueName;
+    if ((event->Par2 <= 0 || event->Par2 >= VARS_PER_TASK) && event->Par1 - 1 != INVALID_TASK_INDEX && GetArgv(Line, valueName, 3))
+    {
+      byte tmpVarNr = findDeviceValueIndexByName(valueName, event->Par1 - 1);
+      if (tmpVarNr != VARS_PER_TASK) {
+        event->Par2 = tmpVarNr + 1;
```

Comment:
Shouldn't this also check the nr. of taskvalues for the specified task?

Reply:
Yes, that would be more correct. Though this won't break anything. I'll have a look on adding that.

Reply:
A few lines below there is a call to `validTaskVars` but that's also not checking for the number of values defined, and neither `taskValueSet` checks this... (but AFAIR there is a check somewhere in the code, I'll just have to find it 😃)